azurerm/resource_arm_storage_share.go

```diff
@@ -75,16 +81,18 @@ func resourceArmStorageShareCreate(d *schema.ResourceData, meta interface{}) err
 	}
 	reference.SetProperties(options)
 
-	d.SetId(name)
+	d.SetId(fmt.Sprintf("%s/%s/%s", name, resourceGroupName, storageAccountName))
 	return resourceArmStorageShareRead(d, meta)
 }
 
 func resourceArmStorageShareRead(d *schema.ResourceData, meta interface{}) error {
 	armClient := meta.(*ArmClient)
 	ctx := armClient.StopContext
 
-	resourceGroupName := d.Get("resource_group_name").(string)
-	storageAccountName := d.Get("storage_account_name").(string)
+	id := strings.Split(d.Id(), "/")
+	name := id[0]
+	resourceGroupName := id[1]
```

Comment:
as discussed on Slack - there's a crash here until we have a state migration for the ID switch